reframe/core/schedulers/__init__.py

```diff
@@ -222,12 +222,22 @@ def sched_exclusive_access(self):
     def prepare(self, commands, environs=None, **gen_opts):
         environs = environs or []
         if self.num_tasks <= 0:
+            num_tasks_per_node = (self.num_tasks_per_node if
+                                  self.num_tasks_per_node else 1)
+            min_num_tasks = (abs(self.num_tasks) if self.num_tasks < 0 else
+                             num_tasks_per_node)
             try:
-                self._num_tasks = (self.guess_num_tasks(abs(self.num_tasks)) if
-                                   self.num_tasks != 0 else
-                                   self.guess_num_tasks())
-                getlogger().debug('flex_alloc_tasks: setting num_tasks to %s' %
-                                  self._num_tasks)
+                guessed_num_tasks = self.guess_num_tasks()
+                if guessed_num_tasks >= min_num_tasks:
+                    self._num_tasks = guessed_num_tasks
+                    getlogger().debug('flex_alloc_tasks: setting num_tasks '
+                                      'to %s' % self._num_tasks)
+                else:
+                    raise JobError(
+                        'not enough nodes satisfying the minimum '
+                        'number of tasks required: %s < %s' %
```

Comment:
I would simply say:

`could not find enough nodes: required X, found Y`

But you should better convert the num tasks you get to node counts.